src/app/component/shared/components/calendar-base/calendar-base.component.ts

```diff
@@ -193,9 +211,41 @@ export class CalendarBaseComponent implements OnInit, OnDestroy {
     this.buildCalendar();
   }
 
+  public formatDate(isMonthCalendar: boolean, dayItem) {
+    if (isMonthCalendar) {
+      return `${dayItem.year}-${ dayItem.month + 1 < 10 ?
+        '0' + (dayItem.month + 1) : dayItem.month + 1}-${dayItem.numberOfDate < 10 ?
+        '0' + dayItem.numberOfDate : dayItem.numberOfDate}`;
+    } else {
+      return `${dayItem.date.getFullYear()}-${ dayItem.date.getMonth() + 1 < 10 ?
+        '0' + (dayItem.date.getMonth() + 1) : dayItem.date.getMonth() + 1}-${dayItem.date.getDate() < 10 ?
+        '0' + dayItem.date.getDate() : dayItem.date.getDate()}`;
+    }
+  }
+
+  public markCalendarDays(isMonthCalendar, days) {
+    days.forEach(day => {
```

Comment:
consider using the map? rxjs method to send requests for each element of the array. You can later call concatAll() or mergeAll(), depending on what you need.